### DISCUSSION TITLE
How to use the Intermediate Representation (IR) of a Rego module to evaluate data

### DISCUSSION THREAD

**Discussion by tomsfernandez** [open-policy-agent/community#400] [OPA and Rego]
I'm interested on keeping the IR of some rego modules that do not change stashed. I hope this will improve the performance of my evaluation.
The thing is, I couldn't find examples in the documentation that show how this is done, can someone give me a hand please ?

**Comment by anderseknert** [comment]
Hi @tomsfernandez 👋 The IR is created via a separate compilation step (`opa build --target plan ...`) for further processing (compilation / transpilation / evaluation..) by another service or application. OPA itself does not use the IR for evaluation.

If you have performance issues, perhaps we could try and figure out what's causing them? Some more details would be helpful if so :)

**Comment by tomsfernandez** [comment]
Hi @anderseknert sorry for taking so long to answer. 
We are using OPA in an application that runs both in a Javascript and in a GoLang environment. We thought caching the IR was a good idea as it was a language agnostic solution.

I'll explore to pre-compile the policy into WASM and see if that improves the performance of my policy. Our rego policy is 2083 lines long (I don't know if that is a lot or not).

Also: is there a way to benchmark how much time time it takes to evaluate every single rule of a rego policy?

**Reply by anderseknert** [reply]
Wasm sounds like the option for reach to in that case, but I'd prefer solving the performance issue first, and then consider if it's actually necessary. Running `opa eval --profile ...` is usually a good place to start.
